README.md

```diff
@@ -1,264 +1,483 @@
 Conjur conjur = new Conjur(credentials);
 ```
 
 ### Authorization Token
+
+```bash
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur endpoint URL>
+# Optional path for non-standard authenticators (e.g. `$CONJUR_APPLIANCE_URL/authn-k8s/myauthenticator`)
+# export CONJUR_AUTHN_URL="<authenticator authn url>"
+```
+
 ```java
+import net.conjur.api.Conjur;
+import net.conjur.api.Token;
+
 Token token = Token.fromFile(Paths.get('path/to/conjur/authentication/token.json'));
 Conjur conjur = new Conjur(token);
 ```
 
 Alternatively, use the `CONJUR_AUTHN_TOKEN_FILE` environment variable:
 ```bash
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur endpoint URL>
+# Optional path for non-standard authenticators (e.g. `$CONJUR_APPLIANCE_URL/authn-k8s/myauthenticator`)
+# export CONJUR_AUTHN_URL="<authenticator authn url>"
 export CONJUR_AUTHN_TOKEN_FILE="path/to/conjur/authentication/token.json"
 ```
 ```java
+import net.conjur.api.Conjur;
+import net.conjur.api.Token;
+
 Token token = Token.fromEnv();
 Conjur conjur = new Conjur(token);
 ```
 
-## Conjur Services Operation Examples
-This section provides some example operations on how invoke the Conjur Java API in your app. 
+## Client APIs
 
-The most common use case is adding and retrieving a secret from Conjur. You can add secrets by invoking our API in the following way: 
+To use the client, you will first create an instance of the client and then call methods
+to send requests to the Conjur API. The most common use case is adding and retrieving
+a secret from Conjur, so we provide some sample code for this use case below.
 
+### Conjur Client Instance (`net.conjur.api.Conjur`)
+
+The client can be instantiated with any of these methods:
 ```java
-conjur.variables().addSecret(VARIABLE_KEY, VARIABLE_VALUE);
+Conjur client = Conjur();
+Conjur client = Conjur(String username, String password);
+Conjur client = Conjur(String username, String password, String authnUrl);
+Conjur client = Conjur(Credentials credentials);
+Conjur client = Conjur(Token token);
 ```
 
-Alternatively, you can add a value to a secret in the Conjur CLI like so: 
-
-`conjur variable values add <NAME_OF_POLICY/VARIABLE> <VALUE_OF_SECRET>`. 
+### Variables (`client.variables()`)
 
-Example: `conjur variable values add policyA/secret-var superSecret`
+#### `void addSecret(String variableId, String secret)`
 
-_NOTE:_ For a variable to be set, it first needs to be created by a policy in Conjur otherwise this operation will fail. To do so, you will need a policy that resembles the one supplied in the [Configuration](#configuration) section above.
+Sets a variable to a specific value based on its ID.
 
-To retrieve this secret:
+Example:
 ```java
-conjur.variables().retrieveSecret("<POLICY_PATH/VAR_NAME>");
-``` 
+import net.conjur.api.Conjur;
+
+Conjur conjur = new Conjur();
+conjur.variables().addSecret(VARIABLE_ID, VARIABLE_VALUE);
```

Comment:
I assume this should be checking the return value ensure the operation succeeded?

Reply:
It doesn't return anything - it throws an error on failure (afaik)

Reply:
Ah, so should it be in a try-catch?  No need to do it now if you don't know.  Just feels like our examples should set a good example.